system/libraries/Session/drivers/Session_database_driver.php

```diff
@@ -89,7 +89,7 @@ public function __construct(&$params)
 		{
 			throw new Exception('Query Builder not enabled for the configured database. Aborting.');
 		}
-		elseif ($this->_db->pconnect)
+		elseif (! $this->_db->pconnect)
```

Comment:
This is a core system change and should be brought up with CodeIgniter directly. I won't make changes to core system files in this repo. 

However, the logic that you're implementing doesn't match the exception that's being thrown so I get the feeling the issue may be elsewhere. Looking at [the docs](http://www.codeigniter.com/user_guide/libraries/sessions.html), it specifically says, "You can NOT use a persistent connection." for the Database driver. If I remember correctly, this was a security issue it might hold on to the session allowing for replay attacks.